src/detectors/threads/index.js

```diff
@@ -11,7 +11,8 @@
  * limitations under the License.
  */
 
-export default async function() {
+export default async function(module, validate) {
+  if (!(await validate(module))) return false;
```

Comment:
It should never be the detector's responsibility to call `validate()`. Can you move this back into the code generator?

Reply:
Actually, you know what? I retract this. You are right, this is better :D